src/main/java/bowling/domain/Players.java

```diff
@@ -0,0 +1,22 @@
+package bowling.domain;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class Players {
```

Comment:
Player 객체의 validateName() 메서드 내부 동작을 보면
String.matches를 통해 이름이 올바른 형식인지 정규표현식을 사용하고 있습니다.
```java
public class Player {
    private static final int NAME_MAX_LENGTH = 3;
    private static final String PATTERN = "^[a-zA-Z]*$";
    private final String name;

    public Player(String name) {
        validateName(name);
        this.name = name;
    }

    private void validateName(String name) {
        if (name.length() != NAME_MAX_LENGTH || !Pattern.matches(PATTERN, name)) {
            throw new IncorrectPlayerNameException();
        }
    }
}
```

String.matches 내부 구현을 보면 
```
// String.matches
public boolean matches(String regex) {
        return Pattern.matches(regex, this);
}

// Pattern
public static boolean matches(String regex, CharSequence input) {
        Pattern p = Pattern.compile(regex);
        Matcher m = p.matcher(input);
        return m.matches();
}
```
매번 계산시 성능이 비싼 Pattern 인스턴스를 생성하고 Pattern 인스턴스는 한번 쓰고 버려지기 때문에 
성능적으로 좋지 않습니다. 뿐더러 자바 doc에서도 한번 컴파일 후 재사용 할것을 권장하고 있습니다. 

If a pattern is to be used multiple times, compiling it once and reusing
it will be more efficient than invoking this method each time

불필요한 객체 생성을 피하도록 개선하면 좋을 것 같아요 :)

Reply:
감사합니다!! 👍👍